### PR TITLE
Alpha 3

### PR DESCRIPTION
**Pull request by luzzif**
Alpha 3 includes bugfixes, full xDAI support, and gathering liquidity from Uniswap/Sushiswap on mainnet, Honeyswap on xDAI with no additional gas fees.

### CODE REVIEW

**Review by PatMacalister — APPROVED** [review]

File: src/constants/index.ts [hunk 1]
```diff
@@ -28,30 +18,31 @@ export const MKR = new Token(ChainId.MAINNET, '0x9f8F72aA9304c8B593d555F12eF6589
...
-
 // used to construct intermediary pairs for trading
 export const BASES_TO_CHECK_TRADES_AGAINST: ChainTokenList = {
-  ...WETH_ONLY,
-  [ChainId.MAINNET]: [...WETH_ONLY[ChainId.MAINNET], DXD[ChainId.MAINNET], DAI, USDC, USDT, COMP, MKR]
+  [ChainId.MAINNET]: [WETH[ChainId.MAINNET], DXD[ChainId.MAINNET], DAI, USDC, USDT, COMP, MKR],
+  [ChainId.RINKEBY]: [WETH[ChainId.RINKEBY]],
+  [ChainId.ARBITRUM_TESTNET_V3]: [Token.WETH[ChainId.ARBITRUM_TESTNET_V3]],
+  [ChainId.SOKOL]: [WSPOA[ChainId.SOKOL]],
+  [ChainId.XDAI]: [WXDAI[ChainId.XDAI]]
```
Comment (jpkcambridge) [inline src/constants/index.ts:27]: Should we add Honey token, and maybe otheres?

File: src/hooks/useSwapCallback.ts [hunk 2]
```diff
@@ -53,7 +53,7 @@ function useSwapCallArguments(
   return useMemo(() => {
     if (!trade || !recipient || !library || !account || !chainId || !deadline) return []
 
-    const contract: Contract | null = getRouterContract(chainId, library, account)
+    const contract: Contract | null = getRouterContract(chainId, library, trade.platform, account)
```
Comment (jpkcambridge) [inline src/hooks/useSwapCallback.ts:56]: Does the library being passed in need to be chain specific?

File: src/index.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 // should always be imported and executed first
-import './set-mainnet-factory-address'
```
Comment (jpkcambridge) [inline src/index.tsx:2]: Why is this mainnet address being removed?

File: src/pages/GovPages/constant.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Currency } from 'dxswap-sdk'
+import { DAI, USDC, USDT, COMP, MKR, AMPL, WBTC } from '../../constants'
+
+export const MainPage = 'Governance Main Page'
+export const PairPage = 'Governance Pair Page'
+
+export const temporaryCurrencyData: Array<Currency> = [DAI, USDC, USDT, COMP, MKR, AMPL, WBTC]
```
Comment (jpkcambridge) [inline src/pages/GovPages/constant.ts:7]: What are these tokens used for?

File: src/utils/maxAmountSpend.ts [hunk 5]
```diff
@@ -5,13 +5,13 @@ import { MIN_ETH } from '../constants'
  * Given some token amount, return the max that can be spent of it
  * @param currencyAmount to return max of
  */
-export function maxAmountSpend(currencyAmount?: CurrencyAmount): CurrencyAmount | undefined {
-  if (!currencyAmount) return undefined
+export function maxAmountSpend(currencyAmount?: CurrencyAmount, chainId?: number): CurrencyAmount | undefined {
+  if (!currencyAmount || !chainId) return undefined
   if (currencyAmount.currency === ETHER) {
```
Comment (jpkcambridge) [inline src/utils/maxAmountSpend.ts:10]: Should this always be checking for ETHER?  Or should it be chain specific?

**Review by jpkcambridge — APPROVED** [review]

**Pull request merged**